packages/sdk/src/utils/encoders.ts

```diff
@@ -1,48 +1,36 @@
-import {Hex, keccak256} from 'viem'
+import {Address, Hex, encodePacked, keccak256} from 'viem'
 import {bytecodes} from '../constants/bytecodes'
 
 import {remove0x, sha3} from './transformers'
-import {BytecodeType, DeploymentConfig} from './types'
+import {BytecodeType, DeploymentConfig, DeploymentConfigSettings} from './types'
 
-export function create2address(deploymentConfig: DeploymentConfig, factoryAddress: string): string {
-  const configHash: string = deploymentConfigHash(deploymentConfig)
-  const futureAddress: string =
-    '0x' +
-    sha3(
-      '0xff' + remove0x(factoryAddress) + remove0x(configHash) + remove0x(sha3(bytecodes[BytecodeType.Holographer])),
-    ).slice(26)
-  return futureAddress
+export function create2address(deploymentConfig: DeploymentConfigSettings, factoryAddress: Address): Address {
+  const configHash: Hex = getErc721DeploymentConfigHash(deploymentConfig.config, deploymentConfig.signer)
+
+  return create2AddressFromDeploymentHash(configHash, factoryAddress)
 }
 
-export function deploymentConfigHash(deploymentConfig: DeploymentConfig): string {
-  const configHash: string = sha3(
-    (
-      '0x' +
-      remove0x(BigInt(deploymentConfig.config.contractType).toString(16)).padStart(64, '0') +
-      remove0x(BigInt(deploymentConfig.config.chainType).toString(16)).padStart(8, '0') +
-      remove0x(BigInt(deploymentConfig.config.salt).toString(16)).padStart(64, '0') +
-      remove0x(sha3(deploymentConfig.config.byteCode)) +
-      remove0x(sha3(deploymentConfig.config.initCode)) +
-      remove0x(BigInt(deploymentConfig.signer).toString(16)).padStart(40, '0')
-    ).toLowerCase(),
-  )
-  return configHash
-} //TODO: validate
+export function create2AddressFromDeploymentHash(configHash: Hex, factoryAddress: Address): Address {
+  return `0x${keccak256(
+    `0xff${remove0x(factoryAddress)}${remove0x(configHash)}${remove0x(keccak256(bytecodes.Holographer))}`,
+  ).slice(26)}`
+}
 
-// deploymentConfigHash(deploymentConfig: DeploymentConfig): string {
```

Comment:
Don't we still need this?

Reply:
I changed the function name, now it's called `getErc721DeploymentConfigHash`